File: examples/akka-quickstart-java/src/main/java/com/example/AkkaQuickstart.java
package com.example;

import akka.actor.typed.ActorSystem;

public class AkkaQuickstart {

    public static void main(String[] args) {
        final ActorSystem<GreeterMain.SayHello> greeterMain =
            ActorSystem.create(GreeterMain.create(), "helloakka");

        greeterMain.tell(new GreeterMain.SayHello("Charles"));

        terminate(greeterMain);
    }

    private static void terminate(ActorSystem system) {
        try {
            Thread.sleep(1000);
            System.out.println(">>> Press ENTER to exit <<<");
            System.in.read();
        }
        catch (Exception ignored) {
            /* ignored */
        }
        finally {
            system.terminate();
        }
    }

}

File: examples/akka-quickstart-scala/src/main/scala/com/example/AkkaQuickstart.scala
package com.example

import akka.actor.typed.ActorSystem

object AkkaQuickstart {

  def main(args: Array[String]): Unit = {
    val greeterMain: ActorSystem[GreeterMain.SayHello] =
      ActorSystem(GreeterMain(), "AkkaQuickStart")

    greeterMain ! GreeterMain.SayHello("Charles")

    terminate(greeterMain)
  }

  private def terminate[T](system: ActorSystem[T]): Unit =
    try {
      Thread.sleep(1000)
      println(">>> Press ENTER to exit <<<")
      System.in.read()
    }
    catch {
      case _: java.io.IOException => /* ignored */
    }
    finally {
      system.terminate()
    }

}
